activerecord/migrations_default_expression.md

```diff
@@ -0,0 +1,51 @@
+# Using an expression as default value for a column with Active Record Migrations
+
+In relational databases, it's a quite common task to put some value in a column that is the result of an expression at the moment of the insertion. It's very useful to get the next sequential number (IDs) or get the current timestamp among others. Using raw SQL we can achieve the latter doing something like:
+
+```sql
+CREATE TABLE timestamps (
+    a_timestamp timestamp without time zone DEFAULT now() NOT NULL
+);
+```
+
+But how do we do it using ActiveRecord's migrations?
+If we try with:
+
+```ruby
+create_table :timestamps do |t|
+  t.timestamp :a_timestamp, null: false, default: 'NOW()'
+end
+```
+
+The `NOW()` will be evaluated at creation time, and will be translated into:
+
+```sql
+CREATE TABLE timestamps (
+    a_timestamp timestamp without time zone DEFAULT '2017-02-03 13:48:43.796235' NOT NULL
+);
+```
+
+Or whatever current timestamp was at that point.
+
+## The solution
+
+The solution for Rails < 5 was to manually `ALTER` the table in the migration:
+
+```ruby
+create_table :timestamps do |t|
+  t.timestamp :a_timestamp, null: false
+end
+execute 'ALTER TABLE ONLY timestamps ALTER COLUMN a_timestamp SET DEFAULT NOW()'
+```
+
+But since Rails 5, we can achieve this much more prettier without writing any database-specific code, using a proc in the `default` attribute:
```

Comment:
Can I suggest an alternative for this line:

```
But starting with Rails 5, we can achieve this within the migration itself, using a proc in the `default` attribute:
```

(I'd drop mention of 'database-specific' in this context, because using a function as the default value is itself database-dependent.